src/Diagnostics/cfl.jl

```diff
@@ -69,34 +73,39 @@ julia> dcfl(model)
 DiffusiveCFL(Δt) = CFL(Δt, cell_diffusion_timescale)
 
 #####
-##### Accurate CFL via reduction
+##### Cell advection time-scale calculation
 #####
 
-using CUDA, CUDAKernels, KernelAbstractions, Tullio
+function cell_advection_timescale(grid, velocities)
+    u, v, w = velocities 
+    min_timescale = 1 / maximum(abs(u) / Δx + abs(v) / Δy + abs(w) / Δz)
```

Comment:
The difference between here and the `tullio` implementation is that here there is an _implicit_ interpolation while the previous version does not interpolate

Reply:
The correct implementation should be without interpolation on a C grid. I am not entirely sure on why... i can check some references

Reply:
Hmm ok! Just to clarify the Tullio implementation didn't interpolate (explicitly or implicitly). Instead it assumed it was valid to add objects at different locations, I guess. I think that's what you meant though by "implicit interpolation".

Reply:
I guess this means that we need abstract operations to be a little less strict 😂

Reply:
I think we can build it manually and avoid some of the auto-interpolating stuff.